src/main/java/ai/elimu/model/contributor/ContributionEvent.java

```diff
@@ -0,0 +1,95 @@
+package ai.elimu.model.contributor;
+
+import ai.elimu.model.BaseEntity;
+import ai.elimu.model.enums.Platform;
+import java.util.Calendar;
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Parent class for various types of content contribution events.
+ */
+@MappedSuperclass
+public class ContributionEvent extends BaseEntity {
+    
+    @NotNull
+    @ManyToOne
+    private Contributor contributor;
+    
+    /**
+     * The content's revision number (after being created/edited).
+     */
+    @NotNull
+    private Integer revisionNumber;
+
+    /**
+     * A comment explaining the contribution.
+     */
+    @Column(length = 1000)
+    private String comment;
+    
+    @NotNull
+    @Temporal(TemporalType.TIMESTAMP)
+    private Calendar time;
+    
+    /**
+     * The time passed during the creation/editing.
+     */
+    @NotNull
+    private Long timeSpentMs;
+    
+    /**
+     * The platform that the {@link Contributor} used when making the contribution.
+     */
+    @NotNull
```

Comment:
Should there be a database migration script for adding the platform to existing contribution events?